SharpDXtest/Tools/ContentBrowserControll.xaml.cs

```diff
@@ -0,0 +1,193 @@
+            }
+        } // Путь к стартовой папке
+        private bool loaded;
+        private List<LogMessage> newMessages = new ();
+
+        public ContentBrowserControl()
+        {
+            InitializeComponent();
+        }
+        
+        private void LoadDirectories(string rootPath, FileItem parent = null)
+        {
+            try
+            {
+                var directories = Directory.GetDirectories(rootPath);
+
+                foreach (var directory in directories)
+                {
+                    var directoryItem = new FileItem { Name = Path.GetFileName(directory), FullPath = directory, Type = ItemType.Folder, Parent = parent };
+                    parent?.Children.Add(directoryItem);
+                    LoadDirectories(directory, directoryItem);
+                }
+            }
+            catch (UnauthorizedAccessException)
+            {
+                // Обработка случаев, когда у пользователя нет доступа к папке
+            }
+        }
+        private void LoadDirectoriesAndFiles(string rootPath, FileItem parent)
+        {
+            try
+            {
+                var directories = Directory.GetDirectories(rootPath);
+
+                foreach (var directory in directories)
+                {
+                    var directoryItem = new FileItem { Name = Path.GetFileName(directory), FullPath = directory, Type = ItemType.Folder, Parent = parent };
+                    parent?.Children.Add(directoryItem);
+                }
+
+                var files = Directory.GetFiles(rootPath);
+                AssetImportContext importContext2 = new AssetImportContext(rootPath);
+                importContext2.LoadAssetMeta();
+                
+                foreach (var file in files)
+                {
+                    if (file.Contains(".meta"))
+                    {
+                        AssetMeta assetMeta = YamlManager.LoadFromFile<AssetMeta>(file);
+                        foreach (var subAsset in assetMeta.SubAssets)
+                        {
+                            var assetsItem = new FileItem { Name = subAsset.Key.Item2, FullPath = file, Type = ItemType.Assets, Parent = parent };
+                            parent?.Children.Add(assetsItem);
+                        }
+                        string text = file.Replace(".meta", String.Empty);
+                        var fileItem = new FileItem
+                        {
+                            Name = Path.GetFileName(text.Substring(0, text.Length - (text.Length - text.IndexOf('.')))), 
+                            FullPath = text, 
+                            Type = ItemType.File, 
+                            Parent = parent
+                        };
+                        parent?.Children.Add(fileItem);
+                        parent?.Children.Remove(parent.Children.FirstOrDefault(x => x.FullPath == text));
+                    }
+                    else
```

Comment:
if file is not an asset - it should not be in content browser

Reply:
Ok